GoogleForCreators/web-stories-wp
# Media: Backfill audio information for existing videos

## Context



## Summary

Add logic to backfill videos that do not have post meta already set. 

## Relevant Technical Choices

- Fixed issues where videos were pointing to the wrong location in PHP unit tests
- Used human readable state for muted. Sadly true/false and null do not work as null is default value of default. [See](https://github.com/WordPress/wordpress-develop/blob/80ccb3697c4a6f5638498c9789ed2abea1070b98/src/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L470)
- Added migration for existing data. 
- Broke processing into processing poster and processing audio. 

## To-do



## User-facing changes



## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Upload video file without sound via Media section of the WP CMS. 
2. Open the editor. 
3. Wait for poster to appear. 
4. See mute icon appear. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #8499
